src/main/java/org/cactoos/scalar/InheritanceLevel.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.scalar;
+
+import org.cactoos.Scalar;
+
+/**
+ * Calculates number of superclasses between base and derived class.
+ *
+ *  <p>This class is thread safe.
+ *
+ *  <p>
+ *  Result interpretation:
+ *
+ *   0 -> classes are identical. (ex. matching IOException with IOException)
+ *   1 -> single level inheritance. (ex. matching FileNotFoundException with
+ *        IOException)
+ *   2 -> two inheritance levels. (ex. matching FileNotFoundException with
+ *        Exception)
+ *   ...
+ *   999 -> classes are not related.
+ *      (ex. matching FileNotFoundException with RuntimeException)
+ *  </p>
+ *
+ * @author Vedran Vatavuk (123vgv@gmail.com)
+ * @version $Id$
+ * @since 0.30
+ */
+public final class InheritanceLevel implements Scalar<Integer> {
+
+    /**
+     * Classes are identical.
+     */
+    private static final int IDENTICAL = 0;
```

Comment:
@Vatavuk I would remove this constant, it's not really helpful. Just use zero and MAX_VALUE.